aws_lambda_builders/workflow.py

```diff
@@ -229,12 +257,31 @@ def __init__(
         self.architecture = architecture
         self.is_building_layer = is_building_layer
         self.experimental_flags = experimental_flags if experimental_flags else []
+
         self.build_in_source = build_in_source
+        if build_in_source is None:
+            self.build_in_source = self.BUILD_IN_SOURCE_BY_DEFAULT
+        self._validate_build_in_source()
 
         # Actions are registered by the subclasses as they seem fit
         self.actions = []
         self._binaries = {}
 
+    def _validate_build_in_source(self):
+        """
+        Validates that the value of build_in_source is supported by the chosen workflow.
+        """
+        if self.build_in_source not in self.BUILD_IN_SOURCE_SUPPORT.value:
+            error_reason = ""
+            if self.BUILD_IN_SOURCE_SUPPORT == BuildInSourceSupport.NOT_SUPPORTED:
+                error_reason = "This workflow does not support building in source"
+            elif self.BUILD_IN_SOURCE_SUPPORT == BuildInSourceSupport.EXCLUSIVELY_SUPPORTED:
+                error_reason = "This workflow only supports building in source"
+            else:
+                error_reason = "Unsupported value for build_in_source"
```

Comment:
Can we simplify this logic?

Reply:
Hmm it could be something like this:

```suggestion
    def _validate_build_in_source(self):
        """
        Validates that the value of build_in_source is supported by the chosen workflow.
        """
        if self.build_in_source not in self.BUILD_IN_SOURCE_SUPPORT.value:
            error_reason = f"Unsupported value for build_in_source. This workflow supports values {self.BUILD_IN_SOURCE_SUPPORT.value}"
```

I just thought the other messages above had better UX/messaging. Plus, I don't really like to include `build_in_source` as the name of this lambda builders parameter could be different than the user-exposed one on SAM CLI.

Reply:
Maybe it could be: "Unsupported value for building in source. This workflow supports values {self.BUILD_IN_SOURCE_SUPPORT.value}"

Reply:
updated this

Reply:
So we won't fail but will print the warning message? (I couldn't find this code any more 😂)

Reply:
correct + I generalized/simplified the validation messaging stuff